[Feature/Enhancement/Custom] - Auto Loot, Quick Looting in stack and Auto Bank function

omeranha opened the pull request:
# Description

This PR add the best autoloot possible, using your manage loot containers quick loot categories and items filter, can be enabled in config.lua!
![image](https://user-images.githubusercontent.com/53233626/155922816-d918e767-b3ca-4950-83df-72d1cdffe177.png)

Quickloot in stack, up to 30 corpses, enabled using client option.
![image](https://user-images.githubusercontent.com/53233626/155922764-ad0e6931-7806-48bc-a886-8bb546cfb9cd.png)

AutoBank function, the dropped coins from monsters will be automatically deposited to your bank account, can be enabled in config.lua.
![image](https://user-images.githubusercontent.com/53233626/155922982-4fd45c3b-cb73-4f7f-bb31-d51ac4b440fd.png)

## Type of change
  - [X] New feature (non-breaking change which adds functionality)

omeranha committed:
Update iologindata.cpp

omeranha committed:
Revert "Update iologindata.cpp"

This reverts commit 205df465d54f423cf1cca7fe219e0627a24b40ec.

omeranha committed:
Merge branch 'opentibiabr:master' into master

omeranha committed:
Merge branch 'opentibiabr:master' into master

omeranha committed:
Merge branch 'opentibiabr:master' into master

omeranha committed:
Merge branch 'opentibiabr:master' into master

omeranha committed:
Merge branch 'opentibiabr:master' into master

omeranha committed:
Merge branch 'opentibiabr:master' into master

omeranha committed:
Merge branch 'opentibiabr:master' into master

omeranha committed:
Merge branch 'opentibiabr:master' into master

omeranha committed:
first commit

Aerwix commented:
Is this the same code as @Hyresu proposed before?

omeranha commented:
> Is this the same code as @hyresu proposed before?

hi, I got from this pull request: https://github.com/opentibiabr/canary/pull/143

guispiller commented:
As I described on discord channel, just to keep it registered:
Tested on OTBR and on canary main, quick loot in stack mechanic is working 100%, but there's a minor issue on the loot message log.
After looting a stack, the last message that shows on screen is always: "No Loot", even if you looted all corpses on the stack. It should be the message before it: "you looted x corpses".

Example on image below
![unknown](https://user-images.githubusercontent.com/11639062/150646775-4d89015b-2d72-45ea-8417-07660020455a.png)

omeranha committed:
Merge branch 'opentibiabr:master' into master

omeranha committed:
Merge branch 'opentibiabr:master' into master

omeranha committed:
Merge branch 'master' into loot-in-stack

omeranha committed:
autoloot

the best autoloot you ever seen

dudantas committed:
Fix sonar check

omeranha committed:
Merge branch 'opentibiabr:master' into master

omeranha committed:
Merge branch 'master' into loot-in-stack

omeranha committed:
fixes and sonar

now autoloot takes all items, sometimes it wasn't catching (I think)
an commented option trying to avoid lag

omeranha committed:
ok sonar, now go away

omeranha committed:
Merge branch 'master' into loot-in-stack

omeranha committed:
should fix no loot message

omeranha committed:
ohno

omeranha committed:
ohno²

majestyotbr commented:
Tested:
- [X] Autoloot.
- [X] Quickloot in stack.

omeranha committed:
Merge branch 'master' into loot-in-stack

omeranha committed:
adding autobank

omeranha committed:
sonar

omeranha committed:
refactor and fix recursive for stash moving option

Sh3Ns commented:
server crash testing in win 10 and linux , quando mata snake.

Issue in Linux #260

![image](https://user-images.githubusercontent.com/50599143/157169605-210f5bbb-3530-4c7f-8880-deb15a7e9273.png)
![image](https://user-images.githubusercontent.com/50599143/157169649-eed2df88-1d05-48a8-9c7b-8686e682ef75.png)

omeranha committed:
Merge branch 'master' into pr/184

omeranha committed:
should fix crash on some monster corpses

omeranha committed:
Update creature.cpp

Sh3Ns reviewed (approved):
passed [Update creature.cpp](https://github.com/opentibiabr/canary/pull/184/commits/417e0b7bc044ce41e60067dbcbb27dcc9d084097)

omeranha committed:
Merge branch 'master' into pr/184

garaflaks commented:
gold in bank, does not compute in hunt analysis
